### PR TITLE
[screenshot] destroy request when done

### PR DESCRIPTION
**Pull request by nullobsi**
destroy the request when it's finished.

- fixes issue where a single application requesting multiple screenshots does not work

### CODE REVIEW

File: src/screenshot/screenshot.c [hunk 1]
```diff
@@ -57,6 +57,7 @@ static int method_screenshot(sd_bus_message *msg, void *data,
 	// TODO: choose a better path
 	const char path[] = "/tmp/out.png";
 	if (!exec_screenshooter(path)) {
+        xdpw_request_destroy(req);
```
Comment (emersion) [inline src/screenshot/screenshot.c:60]: Style: we use tabs for indentation

Comment (nullobsi) [reply src/screenshot/screenshot.c:60]: oops - my editor wasn't set to tabs, let me fix it

**Comment by emersion** [comment]
I wonder if that's the right thing to do. Right now `xdpw_request_destroy` is only called from the `Close` method handler, and multiple screencast sessions work properly. Are you sure it's not a bug in your client?

**Comment by nullobsi** [comment]
according to the spec, Close should be called to cancel the request and close all dialogs associated with the request. I tried calling close from the client, but it never got rid of the object path. Trying to call the screenshot again led to it trying to set the vtable on an already existing object, and failing.

**Comment by emersion** [comment]
What if the client calls `Close`, but at the same time xdpw destroys the object? This seems racy.

**Comment by nullobsi** [comment]
Close is meant to abort the request. The spec says that after calling close, a Response signal will not be emitted. I don't know how the other desktop portals handle this, though.

**Comment by nullobsi** [comment]
`xdg-desktop-portal-gtk` gets rid of the Request object path even if Close is not called.

**Comment by nullobsi** [comment]
while this does seem to be a client program issue (must use token to ensure unique object path,) it doesn't make sense for such an object to persist even beyond the application closing. In fact, calling Close() does not even work-- xdg-desktop-portal destroys its proxy object before the program gets a chance to call it.

**Comment by dmayle** [comment]
subscribe

**Comment by danshick** [comment]
> ~~subscribe~~
Please use the notification feature rather than spamming the comments.

**Comment by WhyNotHugo** [comment]
The same issue seems to exist for screencasts, a bunch of `/org/freedesktop/portal/desktop/request/1_2191/webrtcXXXXXXX` are left behind too.

**Pull request closed without merging**